src/Tribe/Aggregator.php

```diff
@@ -0,0 +1,194 @@
+		$this->service     = Tribe__Events__Aggregator__Service::instance( $this );
+		$this->pue_checker = new Tribe__PUE__Checker( 'http://tri.be/', 'event-aggregator' );
+	}
+
+	/**
+	 * Get the event-aggregator license key
+	 *
+	 * @return string
+	 */
+	public function get_license_key() {
+		return get_option( 'pue_install_key_event_aggregator' );
+	}
+
+	/**
+	 * Get event-aggregator origins
+	 */
+	public function get_origins() {
+		$origins = array();
+
+		if ( $cached_origins = get_transient( "{$this->cache_group}_origins" ) ) {
+			$origins = $cached_origins;
+		} else {
+			$origins = $this->service->get_origins();
+
+			if ( is_wp_error( $origins ) ) {
+				return $origins;
+			}
+
+			set_transient( "{$this->cache_group}_origins", $origins, 6 * HOUR_IN_SECONDS );
+		}
+
+		// Let's build out the translated text based on the names that come back from the EA service
+		foreach ( $origins as &$origin ) {
+			$origin->text = __( $origin->name, 'the-events-calendar' );
+		}
+
+		return apply_filters( 'tribe_ea_origins', $origins );
+	}
+
+	/**
+	 * Fetches an image from the service and saves it to the filesystem if needed
+	 *
+	 * @param string $image_id EA Image ID
+	 */
+	public function get_image( $image_id ) {
+		$tribe_ea_meta_key = 'tribe_ea_image_id';
+
+		require_once ABSPATH . 'wp-admin/includes/image.php';
+
+		$query = new WP_Query( array(
+			'post_type'   => 'attachment',
+			'post_status' => 'any',
+			'meta_key'    => $tribe_ea_meta_key,
+			'meta_value'  => $image_id,
+		) );
+
+		$upload_dir = wp_upload_dir();
+		$file_info = new stdClass;
+
+		// if the file has already been added to the filesystem, don't create a duplicate...re-use it
+		if ( $query->have_posts() ) {
+			$attachment = reset( $query->posts );
+			$attachment_meta = wp_get_attachment_meta( $attachment->ID );
+
+			$file_info->post_id   = $attachment->ID;
+			$file_info->filename  = basename( $attachment_meta['file'] );
+			$file_info->path      = $upload_dir['basedir'] . DIRECTORY_SEPARATOR . $attachment_meta['file'];
+
+			$filetype = wp_check_filetype( $file_info->filename, null );
+			$file_info->extension = $filetype['ext'];
+
+			return $file_info;
+		}
+
+		// fetch an image
+		$response = $this->service->get_image( $image_id );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		// if the reponse isn't an image then we need to bail
+		if ( ! preg_match( '/image/', $response['headers']['content-type'] ) ) {
+			return $response;
+		}
+
+		$extension = str_replace( 'image/', '', $response['headers']['content-type'] );
+
+		if (
+			preg_match( '/filename="([^"]+)"/', $response['headers']['content-disposition'], $matches )
+			&& ! empty( $matches[1] )
+		) {
+			$filename = $matches[1];
+		} else {
+			$filename = md5( $results['body'] ) . '.' . $extension;
+		}
+
+		$filename = sanitize_file_name( $filename );
+
+		// get the file type
+		$filetype = wp_check_filetype( basename( $filepath ), null );
```

Comment:
I don't think `$filepath` is defined before this point; should it be `$filename` or something else here?